extensions/smallrye-context-propagation/deployment/src/main/java/io/quarkus/smallrye/context/deployment/SmallRyeContextPropagationProcessor.java

```diff
@@ -86,4 +106,239 @@ void build(SmallRyeContextPropagationRecorder recorder,
+                            .add(DotNames.NAMED_INSTANCE, AnnotationValue.createStringValue("value", mpConfigIpName)).done();
+                }
+            }
+        });
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    void createSynthBeansForConfiguredInjectionPoints(BuildProducer<SyntheticBeanBuildItem> syntheticBeanBuildItemBuildProducer,
+            SmallRyeContextPropagationRecorder recorder, BeanDiscoveryFinishedBuildItem bdFinishedBuildItem) {
+        Map<String, ExecutorConfig> executorMap = new HashMap<>();
+        Set<String> unconfiguredExecutorIPs = new HashSet<>();
+        Map<String, ThreadConfig> threadContextMap = new HashMap<>();
+        Set<String> unconfiguredContextIPs = new HashSet<>();
+        for (InjectionPointInfo ipInfo : bdFinishedBuildItem.getInjectionPoints()) {
+            AnnotationInstance namedAnnotation = ipInfo.getRequiredQualifier(DotNames.NAMED_INSTANCE);
+            // only look for IP with @NamedInstance on it because the IP transformation made sure it's there
+            if (namedAnnotation == null) {
+                continue;
+            }
+            // furthermore, we only look for any IP that doesn't have other custom qualifier
+            if (ipInfo.getRequiredQualifiers().stream()
+                    .anyMatch(ann -> !ann.name().equals(DotNames.NAMED_INSTANCE)
+                            && !ann.name().equals(DotNames.ANY)
+                            && !ann.name().equals(DotNames.DEFAULT))) {
+                continue;
+            }
+
+            AnnotationInstance meConfigInstance = Annotations.find(extractAnnotations(ipInfo.getTarget()),
+                    DotNames.MANAGED_EXECUTOR_CONFIG);
+            AnnotationInstance tcConfigInstance = Annotations.find(extractAnnotations(ipInfo.getTarget()),
+                    DotNames.THREAD_CONTEXT_CONFIG);
+
+            // get the name from @NamedInstance qualifier
+            String nameValue = namedAnnotation.value().asString();
+
+            if (meConfigInstance == null && tcConfigInstance == null) {
+                // injection point with @NamedInstance on it but no configuration
+                if (ipInfo.getType().name().equals(DotNames.MANAGED_EXECUTOR)) {
+                    unconfiguredExecutorIPs.add(nameValue);
+                } else {
+                    unconfiguredContextIPs.add(nameValue);
+                }
+                continue;
+            }
+            // we are looking for injection points with @ManagedExecutorConfig/@ThreadContextConfig
+            if (meConfigInstance != null || tcConfigInstance != null) {
+                if (meConfigInstance != null) {
+                    // parse ME config annotation and store in a map
+                    executorMap.putIfAbsent(nameValue,
+                            new ExecutorConfig(meConfigInstance.value("cleared"),
+                                    meConfigInstance.value("propagated"),
```

Comment:
Keep in mind that `meConfigInstance.value("propagated")` will not return the defaut value, i.e. `ThreadContext.ALL_REMAINING`, if not specified at all. Which could be ok if you handle this case later on...

See also `AnnotationInstance.valueWithDefault()`.

Reply:
Unless I am misunderstanding you, this should already be handled. See the constructor of `ExecutorConfig` class - `null` is replaced with default value from the annotation.